kibbeh/src/ui/NativeCheckbox.tsx

```diff
@@ -1,94 +1,121 @@
         onClick={onClick}
         data-testid="toggle-input"
       />
-      <span className={togglerClass} {...props} />
+      <span
+        className={on ? "text-primary-100" : "text-primary-300"}
+        {...props}
+      />
     </label>
   );
 };
 
-// Accepts `on` and `children` props and returns `children` if `on` is true
-const ToggleOn = ({ on, children }: ToggleProps & NativeCheckboxProps) => {
+// small util hook for validation of context
+function useToggle() {
+  return React.useContext(MyToggleContext);
+}
+
+function ToggleOn({ children }: ToggleContext) {
+  const { on } = useToggle();
   return <React.Fragment>{on ? children : null}</React.Fragment>;
-};
+}
 
-// Accepts `on` and `children` props and returns `children` if `on` is false
-const ToggleOff = ({ on, children }: ToggleProps & NativeCheckboxProps) => {
+function ToggleOff({ children }: ToggleContext) {
+  const { on } = useToggle();
   return <React.Fragment>{on ? null : children}</React.Fragment>;
-};
-
-// Accepts `on` and `toggle` props and returns the <Switch /> with those props.
-const ToggleButton = ({ on, toggle }: ToggleProps) => (
-  <Toggle on={on} onClick={toggle} />
-);
-
-const allowedTypes = [ToggleOn, ToggleOff, ToggleButton];
-
-const ToggleHelper = ({ children }: any) => {
-  const [on, setOn] = React.useState(false);
-  const toggle = () => setOn(!on);
+}
 
-  return React.Children.map(children, (child) => {
-    if (allowedTypes.includes(child.type)) {
-      const newChild = React.cloneElement(child as any, { on, toggle });
-      return newChild;
-    }
-    return child;
-  });
-};
+function ToggleButton({ ...props }) {
+  const { on, toggle } = useToggle();
+  console.log({ toggle, on });
+  return <ToggleSwitch on={on} onClick={toggle} {...props} />;
+}
 
-export const NativeCheckBox: React.FC = ({ children }) => {
+export const NativeCheckBox: React.FC<ToggleContext> = ({
+  on,
+  title,
+  toggle,
+  description,
+  ...props
+}) => {
   return (
     <div>
-      <React.Fragment>
-        <ToggleHelper>
-          <ToggleOn>Check my box</ToggleOn>
-          <ToggleOff>Yo</ToggleOff>
-          <ToggleButton />
-        </ToggleHelper>
-        <div>{children}</div>
-      </React.Fragment>
+      <Toggle>
+        <div className="container flex-col justify-between">
+          <div
+            style={{
+              width: "601px",
+              height: "64px",
+              borderRadius: "8px",
+              background: "#151A21",
+              justifyContent: "space-between",
+            }}
+            className="flex mt-4"
+          >
+            <div className="ml-3 mt-2 mb-2 relative">
+              <span
+                id="title"
+                className={
+                  on
+                    ? `text-primary-100 font-bold`
+                    : `text-primary-300 font-bold`
+                }
+              >
+                {console.log("on is: ", on)}
+                {title}
+              </span>
```

Comment:
So, right here I expected `on` === `true` would evaluate this condition to true after the **toggle** fires an event, but that didn't happen.